jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -147,6 +148,10 @@ public SpanContext context() {
 
   @Override
   public void finish() {
+    if (finished) {
```

Comment:
this logic should be in the `finishWithDuration` method, and inside the synchronized block